frontend/http/writer.go

```diff
@@ -82,8 +82,15 @@ func WriteScrapeResponse(w http.ResponseWriter, resp *bittorrent.ScrapeResponse)
 	})
 }
 
-func compact(peer bittorrent.Peer) (buf []byte) {
-	buf = []byte(peer.IP)
+func compact4(peer bittorrent.Peer) (buf []byte) {
+	buf = []byte(peer.IP.To4())
```

Comment:
In the off chance that peer.IP.To4() returns nil, this isn't going to fail, but will provide an invalid response.
Let's panic if it's nil.